src/main/java/org/terasology/dynamicCities/settlements/SettlementEntityManager.java

```diff
@@ -120,19 +121,21 @@
     private Random rng;
 
     private Logger logger = LoggerFactory.getLogger(SettlementEntityManager.class);
+
     @Override
     public void postBegin() {
 
         settlementEntities = settlementCachingSystem.getSettlementCacheEntity();
-        rng = new FastRandom(regionEntityManager.hashCode() & 0x921233);
+        long seed = regionEntityManager.hashCode() & 0x921233;
+        rng = new FastRandom(seed);
 
     }
 
     @Override
     public void update(float delta) {
         if (!settlementCachingSystem.isInitialised()) {
             return;
-        } else if (settlementCachingSystem.isInitialised() && settlementEntities == null) {
+        } else if (settlementEntities == null) {
```

Comment:
How come the other check goes away? More curiosity than anything else

Reply:
Because of the previous line, `if (!settlementCachingSystem.isInitialised()) return`, this check is redundant. I removed it because it came up as a warning.

Reply:
Ah yup. Looks makes sense to me